pxe/common_files/disk_imaging.sh

```
@@ -442,19 +445,77 @@ function create_efi_boot_entry() {
+			candidate_is_distro=true
+			if [ -z "$distro_shim_path" ]; then
+				distro_shim_path="$candidate_shim"
+			fi
+		fi
+
+		# Match shim's lookup order: use BOOT.CSV only when the
+		# architecture-specific file is absent or cannot be decoded.
+		candidate_csv=$(find "$candidate_dir" -maxdepth 1 -type f -iname "$boot_csv_name" -print -quit 2>/dev/null)
+		if [ ! -z "$candidate_csv" ]; then
+			candidate_csv_contents=$(iconv -f UTF-16 -t UTF-8 "$candidate_csv" 2>/dev/null)
+			[ $? -ne 0 ] && candidate_csv=
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)